### ISSUE TITLE
Question about reproducing PANTHER with UNI 2

### ISSUE THREAD

**Issue by YuchengXing99** [mahmoodlab/PANTHER#37]
Hi @andrewsong90 ,

Great work on PANTHER—thanks for sharing it!

I tried reproducing the results on the TCGA-BLCA survival prediction task, following your settings closely. The only change I made was replacing the encoder with UNI-2, which outputs 1536-dimensional features.

However, I noticed that the downstream MLP model tends to overfit quite easily with the new embeddings, leading to poor generalization performance.

Have you encountered a similar issue during training? 

Thanks in advance!

**Comment by andrewsong90** [comment]
Hi @Yucheng-Xing 

Apologies for missing out on this one. 
My initial suspicion is that UNI-2 embedding size makes it easier to overfit for sure.

Were you able to resolve this?

Here's few strategies you can try
- Impose heavier dropout for MLP
- Run PCA on UNI-2 embeddings to dramatically reduce embedding size, while retaining much of variance
- Maybe remove the covariance term (instead keep mean and proportion) from PANTHER embeddings => This reduces the overall WSI embedding size, while not really affecting the overall performance in my experience.

Best,
Andrew

**Comment by YuchengXing99** [comment]
Thanks so much for your answer!
